spec/components/candidate_interface/new_references_needed_component_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::NewReferencesNeededComponent do
+  describe '#render?' do
+    let(:application_form) { create(:completed_application_form) }
+
+    context 'when an application form needs more references' do
+      it 'renders the component' do
+        create(:reference, :requested, application_form: application_form, requested_at: Time.zone.now - 30.days)
```

Comment:
Do you also need to stub `can_add_course_choice?` in this example like you do in the other or freeze Time to a known date? Otherwise would it fail after 18 September?

Reply:
Hmm, let's stub it

Reply:
Done

Reply:
:+1: